apps/api/src/common/slug-generator.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { PrismaService } from '@/prisma/prisma.service'
+import generateEntitySlug, {
+  generateSlugName,
+  incrementSlugSuffix
+} from './slug-generator'
+import { mockDeep } from 'jest-mock-extended'
+
+describe('generateEntitySlug', () => {
+  let prisma
+
+  beforeEach(() => {
+    prisma = mockDeep<PrismaService>()
+  })
+
+  describe('generateSlugName', () => {
+    it('should convert name to slug format', () => {
+      expect(generateSlugName('Hello World')).toBe('hello-world')
+      expect(generateSlugName('Entity with 123')).toBe('entity-with-123')
+      expect(generateSlugName('Special #Name!')).toBe('special--name-')
+    })
+  })
+
+  describe('incrementSlugSuffix', () => {
+    it('should return base slug with `-0` when no suffix is found', () => {
+      const result = incrementSlugSuffix('', 'my-slug')
+      expect(result).toBe('my-slug-0')
+    })
+
+    it('should increment suffix when found', () => {
+      const result = incrementSlugSuffix('my-slug-0', 'my-slug')
+      expect(result).toBe('my-slug-1')
+    })
+
+    it('should handle complex increment cases with carryover', () => {
+      const result = incrementSlugSuffix('my-slug-z', 'my-slug')
+      expect(result).toBe('my-slug-00')
```

Comment:
after `my-slug-1`, the next should be `my-slug-2`

Reply:
Yeah I have kept that way only. This case is different. If received `my-slug-z` then the next would be `my-slug-00`

Reply:
Yes, i overlooked it. my bad.